src/mod_offline_riak.erl

```diff
@@ -184,3 +185,37 @@ maybe_decode_timestamp(?INFINITY) ->
 maybe_decode_timestamp(TS) ->
     usec:to_now(TS).
 
+
+get_personal_data(Username, Server) ->
+    LServer = jid:nodeprep(Server),
+    [{offline, ["timestamp", "from", "to", "packet"], fetch_messages(Username, LServer)}].
+
+fetch_messages(Username, LServer) ->
+    LUser = jid:nodeprep(Username),
+    Keys = read_user_idx(LUser, LServer),
+    User = jid:to_binary({Username, LServer}),
+    Msgs = [fetch_msg(Key, LServer, User) || Key <- Keys],
+    lists:flatten(Msgs).
+
+fetch_msg(Key, LServer, To) ->
+    try
+        {ok, Obj} = mongoose_riak:get(bucket_type(LServer), Key),
+
+        PacketRaw = riakc_obj:get_value(Obj),
+        {ok, Packet} = exml:parse(PacketRaw),
```

Comment:
Why do we parse it?

Reply:
oh, doesnt make sense as im parsing it back again. will remove :)